pages/blog/_posts/announcing-batch-keys.mdx

```diff
@@ -0,0 +1,87 @@
+---
+heading: "Announcing: Batch Keys"
+image: /assets/blog/announcing-batch-keys/featured-image.png
+showSubtitle: true
+subtitle: Batch keys allows developers to group work units by leveraging Inngest's efficient event-matching engine.
+date: 2024-07-25
+author: Charly Poly
+disableCTA: false
+---
+
+We are thrilled to release Batch Keys, a powerful addition to _Event Batching_. This new feature allows developers to group work units by leveraging [Inngest's efficient event-matching engine](https://www.inngest.com/blog/accidentally-quadratic-evaluating-trillions-of-event-matches-in-real-time), unlocking new advanced use cases such as Batch User Notifications or more efficient processing of large volumes of events.
+
+[*Event Batching*](https://www.inngest.com/docs/guides/batching), released a few months ago, has been widely adopted by developers and teams using Inngest to build highly scalable features such as ETL, notification systems, and third-party integrations. This feature enabled applications dealing with considerable throughput of events to group similar events in batches to create aggregates, reducing the number of requests to third-party APIs.
+
+![Without and with Event Batching](/assets/blog/announcing-batch-keys/batch-keys-1.jpg)
+
+Developers quickly started to apply *Event Batching* to product-oriented use cases, such as “Grouping similar notifications”, resulting in added complexity in the Functions' body to partition the group events based on event properties, for example, when sending post likes notifications:
```

Comment:
```suggestion
Developers quickly started to apply *Event Batching* to product-oriented use cases, such as grouping similar notifications. This resulted in added complexity in the functions' body to partition group events based on event properties. For example, when sending post likes notifications:
```

Splitting into three sentences for readability

Reply:
Side note: to me, the beginning of this paragraph signals something good (quick adoption!) but the overall meaning of the paragraph is about the tension that arose.

Maybe rephrasing it further to clearly signal the "curse in disguise" would make sense? Something like:

```suggestion
We were delighted that *Event Batching* gained such an enthusiastic adoption. However, we noticed that developers applied it also to product-oriented use cases, such as grouping similar notifications. This resulted in added complexity in the function body to partition group events based on event properties. For example, sending post likes notifications would look as follows:
```